core/util.c

```diff
@@ -217,3 +217,48 @@ void register_print_function(print_message_function_t* function, int log_level)
     print_message_function = function;
     print_message_level = log_level;
 }
+
+/**
+ * Increment the mixed radix number by one according to the permutation matrix.
+ */
+void mixed_radix_incr(mixed_radix_int_t* mixed) {
+	int i = 0;
+	while (i < mixed->size) {
+		int digit_to_increment = mixed->permutation[i];
+		mixed->digits[digit_to_increment]++;
+		if (mixed->digits[digit_to_increment] >= mixed->radixes[digit_to_increment]) {
+			mixed->digits[digit_to_increment] = 0;
+			i++;
+		} else {
+			return; // All done.
+		}
+	}
+	// If we get here, the number has overflowed. Wrap to zero.
+	mixed->digits[i - 1] = 0;
```

Comment:
This is potentially dangerous. If `mixed->size <= 0`, then this will write `0` to some memory location.

Reply:
What do you suggest doing in such error conditions? Note that C does not have exceptions. This code is meant to be invoked only generated code, but it does end up in scope for people writing reactions, so there could be a security issue.  But what to do?  Silently do nothing? Print an error message and exit? Note that in some embedded applications, printing something doesn't make sense.

Reply:
I think the easiest way would be to use `assert`. This way we could at least verify all pre-conditions for debug builds (Assertions are disabled in release builds). If the assertion does not hold, the program would abort.

Typically, C functions use a return value to indicate errors. For instance, the function could return 1 on success and 0 on failure. This would leave it up to the user to handle the error.